sql/core/src/test/scala/org/apache/spark/sql/CollationSQLExpressionsSuite.scala

```diff
@@ -2319,6 +2319,57 @@ class CollationSQLExpressionsSuite
     )
   }
 
+  // scalastyle:off nonascii
+  test("approx_count_distinct returns correct with collation") {
+
+    case class ACDTestCase(
+      collation: String,
+      input: Seq[String],
+      output: Seq[Row]
+    )
+
+    val testCases = Seq(
+      ACDTestCase("utf8_lcase", Seq("a", "a", "A"), Seq(Row(1))),
+      ACDTestCase("utf8_lcase", Seq("aCfew", "acFEw", "ACFEW"), Seq(Row(1))),
+      ACDTestCase("utf8_lcase", Seq("a"), Seq(Row(1))),
+      ACDTestCase("utf8_lcase", Seq("B"), Seq(Row(1))),
+      ACDTestCase("utf8_lcase", Seq("aCfew", "aCfew", "aCfew", "aCfew", "aCfew"), Seq(Row(1))),
+      ACDTestCase("utf8_lcase", Seq("Aaa", "AAA", "AAA", "aAa", "aAA", "aaa"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("č", "c", "Ć", "C", "Z", "Ž"), Seq(Row(2))),
+      ACDTestCase("UNICODE_CI_AI", Seq("Ž"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("z"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("fAFfewfćČĆfečwćCsŠ", "fAffEWfćČCfeCwcćss"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("fAFfe", "fAFfe", "fAFfe", "fAFfe", "fAFfe"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("fAcfewfćČĆfečwćCsŠ", "fAffEWfćČCfeCwcćss"), Seq(Row(2))),
+      ACDTestCase("UNICODE_CI_AI", Seq("Ćcc", "CCc", "CCC", "ĆĆČ", "CČĆ", "ČcĆ", "Ćčč", "ČČČ",
+        "CĆC", "ććć", "ccc"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("sŠŠ", "SŠS", "SŠS", "ššS", "ššs", "SšS", "ššš", "ŠŠs",
+        "ŠsŠ", "ssŠ", "SSS"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("zŽŽ", "ŽZŽ", "ZŽZ", "ZZZ", "ZžZ", "žžž", "žžZ", "ZžZ",
+        "ŽzZ", "zZZ", "ZZZ"), Seq(Row(1))),
+      ACDTestCase("UNICODE_CI_AI", Seq("zŽŽ", "ŽZŽ", "ZŽZ", "ZZZ", "ZžZ", "žžž", "žžZ", "ZžZ",
+        "ŽzZ", "zZZ", "ZZZ", "ZZŠ"), Seq(Row(2)))
+    )
+
+    testCases.foreach( t => {
+      val insertQuery = s"INSERT INTO t VALUES ${t.input.map(s => s"'${s}'").mkString(", ") }"
+      val testQuery = "SELECT approx_count_distinct(s) FROM t"
+
+      sql(s"CREATE TABLE t(s string collate ${t.collation})")
```

Comment:
let's avoid doing `CREATE TABLE t`, instead - please look at other tests to see how `withTable` should be used for this exact purpose

Reply:
on another note, let's try (if we can) to avoid creating a table in the first place

looking at the example for `HyperLogLogPlusPlus`:
```
      > SELECT _FUNC_(col1) FROM VALUES (1), (1), (2), (2), (3) tab(col1);
```

we can properly test the functionality without using `CREATE TABLE`, so let's do that